executor/write.go

```diff
@@ -1366,6 +1357,46 @@ func (e *UpdateExec) Next(goCtx goctx.Context) (Row, error) {
 	return Row{}, nil
 }
 
+// Next implements the Executor Next interface.
+func (e *UpdateExec) Next(goCtx goctx.Context) (Row, error) {
+	if !e.fetched {
+		err := e.fetchRows(goCtx)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		e.fetched = true
+	}
+
+	return e.exec(goCtx, e.SelectExec.Schema())
+}
+
+// NextChunk implements the Executor NextChunk interface.
+func (e *UpdateExec) NextChunk(goCtx goctx.Context, chk *chunk.Chunk) error {
+	chk.Reset()
+	if !e.fetched {
+		err := e.fetchChunkRows(goCtx)
+		if err != nil {
+			return errors.Trace(err)
+		}
+		e.fetched = true
+	}
+
+	for {
+		row, err := e.exec(goCtx, e.children[0].Schema())
+		if err != nil {
+			return errors.Trace(err)
+		}
+
+		// if row != nil, means there is more data to update,
+		// so continue this loop.
```

Comment:
once "row == nil" there is no more data waiting to be updated, the execution of UpdateExec is finished.